docs/adr/2023-08-18_025-resource-based-api.md

```diff
@@ -0,0 +1,146 @@
+---
+slug: 25
+title: |
+  25. Event-sourced, resource-based API
+authors: []
+tags: [Draft]
+---
+
+## Status
+
+Draft
+
+## Context
+
+- [ADR-3](/adr/3) concluded that a full-duplex communication channels are
+  desirable to interact with a _reactive_ system.
+
+- The Client API communicates several types of messages to clients. Currently
+  this ranges from node-level `PeerConnected`, over head-specific `HeadIsOpen`
+  to messages about transactions like `TxValid`. These messages are all of type
+  `ServerOutput`.
+
+- Current capabilities of the API:
+
+  - Clients can retrieve the whole history of `ServerOutput` messages or
+    opt-out using a query parameter - all or nothing.
+
+  - There is a welcome message called `Greetings` which is always sent, that
+    contains the last `headStatus`.
+
+  - There exists a `GetUTxO` query-like `ClientInput`, which will respond with a
+    `GetUTxOResponse` containing the confirmed UTxO in an open head, or (!) the
+    currently committed UTxO when the head is initializing.
+
+  - While overall `json` encoded, clients can choose choose between `json` or
+    binary (`cbor`) output of `transaction` fields in several of these using a
+    query parameter.
+
+- Many of these features have been added in a "quick and dirty" way, by monkey
```

Comment:
Does this adr assume also changing the "quick and dirty" way of doing things?

Reply:
Yes. I added some notes on how content types and query parameters can lead to a fine grained control and more standard way of configuring responses.